src/pattern-library/components/patterns/CustomizingComponentsPage.js

```diff
@@ -14,51 +14,74 @@ export default function CustomizingComponentsPage() {
-            Use an <strong>unstyled component</strong>, e.g.{' '}
-            <code>LinkUnstyled</code>. This is appropriate if you want to do
-            considerable customization of a component.
-          </li>
-          <li>
-            Use the{' '}
-            <strong>
-              <code>classes</code> prop
-            </strong>
-            . This allows authors to append arbitrary classes to the{' '}
-            {"component's"} outermost element but should be used with
-            consideration.
-          </li>
-          <li>
-            Use a <strong>different component</strong>, or create a new one.
-          </li>
-          <li>
-            Emergency escape hatch: Use{' '}
-            <strong>
-              <code>!important</code> CSS rules
-            </strong>{' '}
-            (these class names are prefixed with a <code>!</code> in
-            <code>tailwindcss</code>) . This should be a last-ditch tactic.
-          </li>
-        </ol>
+      </Library.Pattern>
 
-        <Library.Example title="Stop and reflect">
+      <Library.Pattern title="How to customize a component">
+        <Library.Example title="Intended customization">
+          <p>
+            The following options represent flexibility purposely baked into
+            components.
+          </p>
+          <ol>
+            <li>
+              Use a <strong>different variant</strong> of the component. Check
+              the {"component's"} documentation in this library to see what
+              variants are available.
+            </li>
+            <li>
+              Use <strong>component styling props</strong>, e.g.{' '}
+              <code>color</code>, <code>underline</code> or <code>size</code>.
+              These differ per component and define styling aspects that are
+              intended to be adjusted.
+            </li>
+            <li>
+              Use an <strong>unstyled component</strong>, e.g.{' '}
+              <code>LinkUnstyled</code>. This is appropriate if you want to do
+              considerable customization of a component.
+            </li>
+          </ol>
+        </Library.Example>
+
+        <Library.Example title="Stronger medicine">
+          <p>
+            For use if the above options do not satisfy the need. Tread with
+            care.
+          </p>
+          <ol>
+            <li>
+              <p>
+                Use the{' '}
+                <strong>
+                  <code>classes</code> prop
+                </strong>
+                . This allows authors to append arbitrary classes to the{' '}
+                {"component's"} outermost element but should be used with
+                consideration.
+              </p>
+              <p>
+                The intention is to allow for extension, not override. There is
+                no guarantee that provided <code>classes</code> {"won't"}{' '}
+                conflict with classes applied by the component.
+              </p>
```

Comment:
I would suggest to say something like, "in the event of a conflict between the component's own styling and the classes specified here, there is currently no guarantee of which will take precedence." This issue creates a hazard when updating the library, so it would be good for us to find a general solution to this in future. [tailwind-merge](https://github.com/dcastil/tailwind-merge) is an example of the kind of thing I was thinking, though the implementation of that is unfortunately quite complex.